hecuba_py/src/hfilter.py

```diff
@@ -0,0 +1,268 @@
+from collections import Iterable
+
+import regex
+import inspect
+from hecuba import config
+from hecuba.qbeast import QbeastIterator, QbeastMeta
+
+from IStorage import IStorage
+from hecuba.tools import NamedItemsIterator
+
+magical_regex = regex.compile('(?:\d+(?:\.\d+)?|\w|"\w+")+|[^\s\w\_]')
+
+
+def func_to_str(func):
+    func_string = inspect.getsourcelines(func)[0][0]
+    start, end = func_string.find("lambda"), func_string.rfind(",")
+    func_string = func_string[start:end]
+    func_vars = func_string[7:func_string.find(':')].replace(" ", "").split(',')
+    clean_string = func_string[func_string.find(':') + 1:].replace("\\n", '').replace("'", '')
+    return func_vars, clean_string
+
+
+def substit_var(final_list, func_vars, dictv):
+    list_with_values = []
+    for elem in final_list:
+        if not isinstance(elem, (str, unicode)) and isinstance(elem, Iterable):
+            list_with_values.append(elem)
+        elif (elem != 'in' and not isinstance(elem, int) and not regex.match(r'[^\s\w]', elem)) and not elem.isdigit():
+            i = elem.find('.')
+            if i > 0:
+                elem_var = elem[:i]
+                if elem_var not in func_vars:
+                    elemm = elem[i:]
+                    get_ele = dictv.get(str(elemm))
+                    if get_ele is None:
+                        list_with_values.append(elem)
+                    else:
+                        list_with_values.append(dictv.get(str(elem)))
+                else:
+                    list_with_values.append(elem[i+1:])
+            else:
+                get_elem = dictv.get(str(elem), elem)
+                list_with_values.append(get_elem)
+        else:
+            list_with_values.append(elem)
+
+    return list_with_values
+
+
+def is_float(var):
```

Comment:
```python
is_numerical = regex.compile('\d+(\.\d+)?')

def is_float(var):
    return is_numerical.match(var) is not None

# And maybe we should call this method is_numerical, because accepts ints too
```